sstables/sstables.cc

```diff
@@ -3143,6 +3144,46 @@ std::optional<large_data_stats_entry> sstable::get_large_data_stat(large_data_ty
     return std::make_optional<large_data_stats_entry>();
 }
 
+// The gc_before returned by the function can only be used to estimate if the
+// sstable is worth dropping some tombstones. We only return the maximum
+// gc_before for all the partitions that have record in repair history map. It
+// is fine that some of the partitions inside the sstable does not have a
+// record.
```

Comment:
We could return the weighted average (weighted by the number of tokens). I don't suggest to change anything now.

Reply:
OK. I have took a note.